packages/shared/lib/ledger.ts

```diff
@@ -243,7 +244,7 @@ export function getLegacyErrorMessage(error: any, shouldLocalize: boolean = fals
 }
 
 export function formatAddressForLedger(address: string, removeChecksum: boolean = false): string {
-    if (removeChecksum) {
+    if (address.length === LEGACY_ADDRESS_WITH_CHECKSUM_LENGTH && removeChecksum) {
```

Comment:
This might affect non-legacy transactions in the future if for whatever reason we try to remove its checksum.

Reply:
non-legacy transactions don't have an address length of 90 - it's 64